### DISCUSSION TITLE
Does Zigbee2MQTT require a client certificate to connect to MQTTS?

### DISCUSSION THREAD

**Discussion by switzer60** [Koenkk/zigbee2mqtt#20806] [General]
First off, let me thank the authors of this software and the community that supports it.  Thank you.

I am trying to have zigbee2mqtt connect to an MQTTS service with a username/password combination 

I'm a few hours into using the software.  I've configured an eclipse-mosquitto docker container running on port 8883 with a regular SSL cert signed by global CA (not Let's Encrypt). 

%openssl s_client -connect my.fqdn.com:8883
CONNECTED(00000003)
depth=2 C = BE, O = GlobalSign nv-sa, OU = Root CA, CN = GlobalSign Root CA

I'm able to connect to moquitto with MQTTX with the following:
host: mqtts://my.fqdn.com
port: 8883
Username: my-username
Password: my-secret

From reading the documentation here:  https://www.zigbee2mqtt.io/guide/configuration/mqtt.html, I have placed the following in my configuration.yaml:

mqtt:
  base_topic: zigbee2mqtt
  server: 'mqtt://my.fqdn.com:1883'
  user: my-username
  password: my-secret
  keepalive: 60
  reject_unauthorized: true
  version: 5

This connects fine, everything works as expected. I can connect to frontend and navigate around.  Great!
NOTE: The above yaml connects using MQTT on port 1883.

However, if I place the following in my configuration.yaml:

mqtt:
  base_topic: zigbee2mqtt
  server: 'mqtts://my.fqdn.com:8883'
  user: my-username
  password: my-secret
  keepalive: 60
  reject_unauthorized: true
  version: 5

NOTE: The above yaml intends MQTT-S- on port 8883.

When I start the container I get this in the logs:

Zigbee2MQTT:info  2024-01-14 23:46:50: Logging to console and directory: '/app/data/log/2024-01-14.23-46-50' filename: log.txt
Zigbee2MQTT:info  2024-01-14 23:46:50: Starting Zigbee2MQTT version 1.35.1 (commit #9eaaa0f)
Zigbee2MQTT:info  2024-01-14 23:46:50: Starting zigbee-herdsman (0.30.0)
Zigbee2MQTT:info  2024-01-14 23:46:55: zigbee-herdsman started (resumed)
Zigbee2MQTT:info  2024-01-14 23:46:55: Coordinator firmware version: '{"meta":{"maintrel":"3 ","majorrel":"6","minorrel":"10","product":8,"revision":"6.10.3.0 build 297"},"type":"EZSP v8"}'
Zigbee2MQTT:info  2024-01-14 23:46:55: Currently 0 devices are joined:
Zigbee2MQTT:warn  2024-01-14 23:46:55: `permit_join` set to  `true` in configuration.yaml.
Zigbee2MQTT:warn  2024-01-14 23:46:55: Allowing new devices to join.
Zigbee2MQTT:warn  2024-01-14 23:46:55: Set `permit_join` to `false` once you joined all devices.
Zigbee2MQTT:info  2024-01-14 23:46:55: Zigbee: allowing new devices to join.
Zigbee2MQTT:info  2024-01-14 23:46:55: Connecting to MQTT server at mqtts://my.fqdn.com:8883

And it just "hangs" here, and the frontend doesn't start up.

I'm not really wanting to set up a client certificate, I'm happy with the mqtt client authenticating with a username/password combination over TLS (like I do with MQTTX)

Based on this documentation:
  # Optional: absolute path to SSL/TLS certificate of CA used to sign server and client certificates (default: nothing)
  ca: '/etc/ssl/mqtt-ca.crt'

I can't meet the mqtt-ca.crt criteria, because I can't provide a CA certificate that has signed -BOTH- my server certificate and my client certificate (I'd have to buy a separate client certificate from the CA).  I could have a CA that signed my server certificate and a -different- CA (i.e. self-signed CA) create my client certificate, but I can't see how that would be configured.  Can I have two ca: lines?

In my mind, the expected behavior would be the zigbee2mqtt can connect to an MQTT server with TLS over port 8883 and use a username/password combination to authenticate (like MQTTX).

I'm not sure if I'm doing something wrong and/or if my expected behavior is wrong.

I'd appreciate any help!

Thank you!

**Comment by wveilleux** [comment]
I'm trying about the same thing with the same issue with an Azure IoT Hub with mqtt on port 8883 with TLS. I'm just wondering if the SSL/TLS crypto library support TLS v1.2 or only v1.0.
@switzer60 , did you find out a solution ?

**Comment by switzer60** [comment]
Thanks for giving me a poke to update my question.  I got it working.

My configuration.yaml looks like this now:

mqtt:
  base_topic: zigbee2mqtt
  server: mqtts://<full.fqdn.here>
  user: <username here>
  password: <password here>
  version: 5


It must have been a problem with those tick marks, and no port number?

Please post back if this works for you and then we can mark it as the solution.
